src/checkpoints/checkpoints.cpp

```diff
@@ -200,8 +200,20 @@ namespace cryptonote
       }
     }
 
-    uint64_t const end_cull_height = height - service_nodes::CHECKPOINT_STORE_PERSISTENTLY_INTERVAL;
-    uint64_t start_cull_height     = (end_cull_height < service_nodes::CHECKPOINT_STORE_PERSISTENTLY_INTERVAL)
+    uint64_t end_cull_height = height - service_nodes::CHECKPOINT_STORE_PERSISTENTLY_INTERVAL;
+    {
+      uint64_t constexpr num_desired_checkpoints = service_nodes::CHECKPOINT_NUM_CHECKPOINTS_FOR_CHAIN_FINALITY + 1;
```

Comment:
Why +1?

Reply:
Hmm. Actually turns out simpler if I remove. Just need to change the culling condition to be < instead of <=

Reply:
Or, well. If you have 3 checkpoints at 8, 12, 16. You can cancel checkpoints 12, 16 and all the blocks from [8->12) are reorganisable, so you only want to cull checkpoints up to height < 8.

Am I thinking that through right?

Reply:
If 8, 12, 16 are on the same chain then nothing before 12 should be reorg possible.  The only time 8 would matter is if 12 and 16 are on different chains (which means, essentially, that 8 is the 2nd-last checkpoint on two different chains).

Reply:
I gotchu. Updated